src/Catalyst.Core.Modules.Dfs.Tests/IntegrationTests/CoreApi/FileSystemApiTest.cs

```diff
@@ -0,0 +1,1013 @@
+            Assert.Equal(false, node.IsDirectory);
+            Assert.Equal(4, links.Length);
+            Assert.Equal("bafkreigwvapses57f56cfow5xvoua4yowigpwcz5otqqzk3bpcbbjswowe", links[0].Id);
+            Assert.Equal("bafkreiew3cvfrp2ijn4qokcp5fqtoknnmr6azhzxovn6b3ruguhoubkm54", links[1].Id);
+            Assert.Equal("bafkreibsybcn72tquh2l5zpim2bba4d2kfwcbpzuspdyv2breaq5efo7tq", links[2].Id);
+            Assert.Equal("bafkreihfuch72plvbhdg46lef3n5zwhnrcjgtjywjryyv7ffieyedccchu", links[3].Id);
+
+            var text = await ipfs.UnixFsApi.ReadAllTextAsync(node.Id);
+            Assert.Equal("hello world", text);
+        }
+
+        [Fact]
+        public async Task Add_Protected()
+        {
+            var options = new AddFileOptions
+            {
+                ProtectionKey = "self"
+            };
+            var node = await ipfs.UnixFsApi.AddTextAsync("hello world", options);
+            Assert.Equal("cms", node.Id.ContentType);
+            Assert.Equal(0, node.Links.Count());
+            Assert.Equal(false, node.IsDirectory);
+
+            var text = await ipfs.UnixFsApi.ReadAllTextAsync(node.Id);
+            Assert.Equal("hello world", text);
+        }
+
+        [Fact]
+        public async Task Add_Protected_Chunked()
+        {
+            var options = new AddFileOptions
+            {
+                ProtectionKey = "self",
+                ChunkSize = 3
+            };
+            var node = await ipfs.UnixFsApi.AddTextAsync("hello world", options);
+            Assert.Equal(4, node.Links.Count());
+            Assert.Equal(false, node.IsDirectory);
+
+            var text = await ipfs.UnixFsApi.ReadAllTextAsync(node.Id);
+            Assert.Equal("hello world", text);
+        }
+
+        [Fact]
+        public async Task Add_OnlyHash()
+        {
+            var nodes = new[]
+            {
+                "QmVVZXWrYzATQdsKWM4knbuH5dgHFmrRqW3nJfDgdWrBjn",
+                "QmevnC4UDUWzJYAQtUSQw4ekUdqDqwcKothjcobE7byeb6",
+                "QmTdBogNFkzUTSnEBQkWzJfQoiWbckLrTFVDHFRKFf6dcN",
+                "QmPdmF1n4di6UwsLgW96qtTXUsPkCLN4LycjEUdH9977d6",
+                "QmXh5UucsqF8XXM8UYQK9fHXsthSEfi78kewr8ttpPaLRE"
+            };
+            foreach (var n in nodes)
+            {
+                await ipfs.BlockApi.RemoveAsync(n, ignoreNonexistent: true);
+            }
+
+            var options = new AddFileOptions
+            {
+                ChunkSize = 3,
+                OnlyHash = true,
+            };
+            var node = await ipfs.UnixFsApi.AddTextAsync("hello world", options);
+            var links = node.Links.ToArray();
+            Assert.Equal(nodes[0], node.Id);
+            Assert.Equal(nodes.Length - 1, links.Length);
+            for (var i = 0; i < links.Length; ++i)
+            {
+                Assert.Equal(nodes[i + 1], links[i].Id);
+            }
+
+            // TODO: Need a method to test that the CId is not held locally.
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)